zerver/views/messages.py

```diff
@@ -319,7 +340,9 @@ def highlight_string_text_offsets(text, locs):
     pos = 0
     result = u''
     for loc in locs:
-        (offset, length) = loc
+        (offset, length) = highlight_match_tag_pos(string, pos, loc, u'<', u'>')
+        if offset < pos:
+            continue
```

Comment:
What is a case where this `continue` is needed?  May be worth including a test for that case if there isn't one already.